### PR TITLE
noisetorch: 0.10.1 -> 0.11.3

### PR DESCRIPTION
**Pull request by legendofmiracles**
###### Motivation for this change
This program only works in combination of the nixos/noisetorch module (and as such only works on nixos), since file capabilities aren't set otherwise - because the newest version hard depends on them.
Can I make the user aware of that, or set them in the build process?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127457` at dd97a2f3 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>noisetorch</li>
  </ul>
</details>

**Comment by legendofmiracles** [comment]
Suppresses the noise for me, on pipewire.

File: pkgs/applications/audio/noisetorch/default.nix [hunk 1]
```diff
@@ -1,42 +1,35 @@
...
-    # Get version from environment instead of git tags
-    ./version.patch
-  ];
-
   vendorSha256 = null;
 
   doCheck = false;
 
-  subPackages = [ "." ];
+  buildFlagsArray = [ "-ldflags=-X main.version=${version} -X main.distribution=nix" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/noisetorch/default.nix:18]: Please use ldflags instead.

File: pkgs/applications/audio/noisetorch/default.nix [hunk 2]
```diff
@@ -1,42 +1,35 @@
...
 
-  buildInputs = [ rnnoise-plugin ];
+  subPackages = [ "." ];
 
-  postPatch = "substituteInPlace main.go --replace 'librnnoise_ladspa/bin/ladspa/librnnoise_ladspa.so' '$RNNOISE_LADSPA_PLUGIN'";
+  buildInputs = [ copyDesktopItems ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/noisetorch/default.nix:22]:
```suggestion
  nativeBuildInputs = [ copyDesktopItems ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/noisetorch/default.nix (hunk 1 above)
Comment (legendofmiracles) [reply pkgs/applications/audio/noisetorch/default.nix]: Thanks for the feedback

**Comment by legendofmiracles** [comment]
@SuperSandro2000 i think the package is broken on non-nixos, since the capabilities aren't set - as i pointed out in the PR description.

Is there are way to set the capabilities at buildtime? Or just set the package for nixos only?

(I probably should have made the PR a draft, but i thought whoever merged the PR would read the comment and then give feedback)

**Pull request merged**